src/pages/en/guides/cms/contentful.md

```diff
@@ -0,0 +1,445 @@
+```
+
+Now, you should be able to use these environment variables in your project. If you would like to have IntelliSense for your Contentful environment variables, you can create a `env.d.ts` file in the `src/` directory and configure `ImportMetaEnv` like this:
+
+```ts title="src/env.d.ts"
+interface ImportMetaEnv {
+  readonly CONTENTFUL_SPACE_ID: string;
+  readonly CONTENTFUL_DELIVERY_TOKEN: string;
+  readonly CONTENTFUL_PREVIEW_TOKEN: string;
+}
+```
+Check out [environment variables](/en/guides/environment-variables/) for more information on how to use them.
+
+Your root directory should now include these new files:
+
+```ini title="" ins={2-3}
+├── src/
+│   └── env.d.ts
+├── .env
+├── astro.config.mjs
+└── package.json
+```
+
+### Installing dependencies
+
+Install the following packages with your favorite package manager:
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  # install contentful and @contentful/rich-text-html-renderer with npm 
+  npm install contentful @contentful/rich-text-html-renderer
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  # install contentful and @contentful/rich-text-html-renderer with pnpm
+  pnpm install contentful @contentful/rich-text-html-renderer
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  # install contentful and @contentful/rich-text-html-renderer with yarn
+  yarn add contentful @contentful/rich-text-html-renderer
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+Next, create a new file called `contentful.ts` in a `src/lib` directory in your project.
+
+```ts title="src/lib/contentful.ts"
+import contentful from "contentful";
+
+export const contentfulClient = contentful.createClient({
+  space: import.meta.env.CONTENTFUL_SPACE_ID,
+  accessToken: import.meta.env.DEV
+    ? import.meta.env.CONTENTFUL_PREVIEW_TOKEN
+    : import.meta.env.CONTENTFUL_DELIVERY_TOKEN,
+  host: import.meta.env.DEV ? "preview.contentful.com" : "cdn.contentful.com",
+});
+
+```
+
+The above code snippet is creating a new contentful client and passing in the credentials from the `.env` file. In development mode (`import.meta.env.DEV = true`), the client will use the preview token and preview host. In production mode, the client will use the delivery token and delivery host.
+
+:::caution
+While in development mode, your content will be fetched from the **Contentful preview API**. This means that you will be able to see unpublished content from the Contentful web app. At build time, your content will be fetched from the **Contentful delivery API**. This means that only published content will be available at build time. 
+:::
+
+Finally, your root directory should now include these new files:
+
+```ini ins={3-4}
+├── src/
+│   └── env.d.ts
+│   └── lib/
+│       └── contentful.ts
+├── .env
+├── astro.config.mjs
+└── package.json
+```
+
+### Fetching data
+
+Now that you have your Contentful client set up, you can fetch data from Contentful inside your Astro components. 
+
```

Comment:
I agree. Maybe we can have it in both sections? I like both ideas and I am not sure which one we should keep the tip

Reply:
We need to avoid the "referencing loop": the tutorial section references the setup in the first section as a prereq, so if we pause the setup in the first section to say "check out the tutorial section", we have a bit of a problem! Try and rephrase the first tip to avoid that